parcels/field.py

```diff
@@ -684,12 +684,17 @@ def from_netcdf(
         if grid is None:
             # Concatenate time variable to determine overall dimension
             # across multiple files
-            time, time_origin, timeslices, dataFiles = cls._collect_timeslices(
-                timestamps, data_filenames, _grid_fb_class, dimensions, indices, netcdf_engine
-            )
-            grid = Grid.create_grid(lon, lat, depth, time, time_origin=time_origin, mesh=mesh)
-            grid.timeslices = timeslices
-            kwargs["dataFiles"] = dataFiles
+            if "time" in dimensions:
```

Comment:
Given the code before didn't error in `_collect_timeslices` (only in the array assignment later), are there times (i.e., in other setups) where "time" isn't in `dimensions`, but the field isn't `CS_w`? Hence adding this clause could cause something else to break? Just want to understand a bit more before merging since I'm not super familiar with this code

Reply:
Yes, I understand your concern. This part of the code is so bloated that it's not easy to keep the oversight. But given that we have failing CI, something is probably going wrong. I'll look into how I can find a different, more robust way to fix this issue

Reply:
Looking forward to using xarray 😁

Reply:
I looked at the code again, and I now _do_ think this is robust code. 

The functionality of the part of the code that I changed is to create a list `timeslices` of the time associated with each snapshot of a Field in a netcdf file. But if a Field doesn't have a time dimension, this operation does not make sense. 

So I'm pretty confident that this PR will have _no_ major unintended consequences; it's just that other Fields without a time dimension never reach this part of the code so it's not an issue for these. 

I could dig much deeper what happens to these fields then, but I suggest we leave that for a major refactor